solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -406,15 +436,15 @@ static Long getCheckSum(Checksum checksum, File f) {
   private volatile IndexFetcher currentIndexFetcher;
 
   public IndexFetchResult doFetch(SolrParams solrParams, boolean forceReplication) {
-    String masterUrl = solrParams == null ? null : solrParams.get(MASTER_URL);
+    String leaderUrl = solrParams == null ? null : solrParams.get(LEADER_URL);
```

Comment:
hmmm I think we need to read `masterUrl` here too and we missed it. I'll do the change

Reply:
I wasn't able to push to the PR directly, don't know if I'm doing something wrong or something wasn't correctly configured. I'll push to master directly (won't show up here)

Reply:
That's weird. All maintainers had edit access. Thanks for making the change and merging the PR. Perhaps, I can open another PR with a test case for this ternary? Would that be valuable since it wasn't caught by the added test.

Reply:
Commented from startup account. 🙃 Do you want me to create a test case for this additional ternary operator?

Reply:
You have maintainer edit rights. Not sure what happened

Reply:
It should be tested with this change: https://github.com/apache/lucene-solr/pull/1711/commits/8ae5a476021fa5ad122375080d315c0234229a2d#diff-a711e0f6e912250c48a73a96d1d01c57R843, but it's only tested "rarely()", which may gave been why we didn't catch in on our tests.

Reply:
(Most likely I did something wrong with my setup, I've never committed to some 3rd party PR before)